guides/hack/11-built-in-types/28-shapes.md

```diff
@@ -209,4 +222,8 @@ function returns_wrong_shape(): shape('x' => int) {
 
 ## Converting Shapes
 
-You can use `Shapes::toArray` to convert a shape to a `darray`.
+You can use [`Shapes::toArray`](https://docs.hhvm.com/hack/reference/class/HH.Shapes/toArray/) to convert a shape to a `darray`, or You can use [`Shapes::toDict`](https://docs.hhvm.com/hack/reference/class/HH.Shapes/toDict/) to convert a shape to a `dict`.
```

Comment:
Dict needs to be preferred , and pre-existing: conversion should be discouraged.

Perhaps:

> Converting shapes to containers is strongly discouraged, however is necessary, this can be done with `Shapes:toDict()`.
>
> On older versions of HHVM, Shapes can also be converted to `darray`s with `Shapes::toArray()`; this should be avoided in new code, as `darray`s are currently an alias for the `dict` type, and will be removed from the language.

Reply:
Also, throughout: do not specify link targets if they're internal. `Foo::bar()` in backticks will be automatically converted to links.